src/shared/components/mutationTable/column/cancerCellFraction/CancerCellFractionElement.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import SampleManager from 'pages/patientView/SampleManager';
+
+const maxBarHeight = 12;
+const barWidth = 6;
+const barSpacing = 3;
+const indexToBarLeft = (n: number) => n * (barWidth + barSpacing);
+
+function getSVGWidth(numberOfSamples: number) {
+    return numberOfSamples * barWidth + (numberOfSamples - 1) * barSpacing;
+}
+
+const CancerCellFractionElementTooltip: React.FunctionComponent<{
+    sampleIds: string[];
+    sampleToCCFValue: any;
+    sampleManager?: SampleManager;
+}> = props => {
+    const sampleOrder = props.sampleManager
+        ? props.sampleManager.getSampleIdsInOrder()
+        : [];
+    const orderedSamplesWithValues = sampleOrder.filter(
+        (sampleId, index, array) => {
+            return props.sampleToCCFValue[sampleId];
+        }
+    );
+    const tooltipLines = orderedSamplesWithValues.map(
+        (sampleId, index, array) => (
+            <span key={sampleId}>
+                {props.sampleManager
+                    ? props.sampleManager.getComponentForSample(sampleId, 1, '')
+                    : null}{' '}
+                {props.sampleToCCFValue[sampleId]}
+                <br />
+            </span>
+        )
+    );
+    return <span>{tooltipLines}</span>;
+};
+
+const CancerCellFractionBar: React.FunctionComponent<{
+    ccfValue: string;
+    color: string;
+    barX: number;
+}> = props => {
+    const barHeight =
+        (isNaN(+props.ccfValue) ? 0 : +props.ccfValue) * maxBarHeight;
```

Comment:
ccfValue is a string, what do you want to do `+props.ccfValue`?

Reply:
@sheridancbio @zhx828 see here https://medium.com/@nikjohn/cast-to-number-in-javascript-using-the-unary-operator-f4ca67c792ce, it casts a string to a number